From 731282ac0516fea7a9db38cda3a0a2dd7bd491a8 Mon Sep 17 00:00:00 2001
From: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
Date: Mon, 3 Jun 2013 13:16:05 -0700
Subject: sata_rcar: fix compilation warning in sata_rcar_thaw()

When compiling the driver with gcc 4.8, it gives the following warning:

 drivers/ata/sata_rcar.c: In function `sata_rcar_thaw':
 drivers/ata/sata_rcar.c:183:2: warning: large integer implicitly truncated to unsigned type [-Woverflow]

Fix the warning by explicit cast of the 'unsigned long' value to 'u32'.

Signed-off-by: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
Signed-off-by: Tejun Heo <tj@kernel.org>
(cherry picked from commit 5ba59b59cb413b9d89f40532bad3529d5185dd3c)
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>

Conflicts:
	drivers/ata/sata_rcar.c
---
 drivers/ata/sata_rcar.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/ata/sata_rcar.c b/drivers/ata/sata_rcar.c
index b1dd5d82..8108eb06 100644
--- a/drivers/ata/sata_rcar.c
+++ b/drivers/ata/sata_rcar.c
@@ -186,7 +186,7 @@ static void sata_rcar_thaw(struct ata_port *ap)
 	void __iomem *base = priv->base;
 
 	/* ack */
-	iowrite32(~SATA_RCAR_INT_MASK, base + SATAINTSTAT_REG);
+	iowrite32(~(u32)SATA_RCAR_INT_MASK, base + SATAINTSTAT_REG);
 
 	ata_sff_thaw(ap);
 
-- 
1.8.4.3.gca3854a

